core/src/components/datetime/utils/format.ts

```diff
@@ -24,7 +26,11 @@ export const getFormattedTime = (refParts: DatetimeParts, use24Hour: boolean): s
     return `${hour}:${minute}`;
   }
 
-  return `${hour}:${minute} ${getFormattedAMPM(refParts.ampm)}`;
+  return new Intl.DateTimeFormat(locale, {
```

Comment:
Any reason why we use `Intl.DateTimeFormat` here but not when we have 24 hour time above?